ui/src/logs/actions/index.ts

```diff
@@ -51,12 +85,73 @@ export const setSource = (source: Source): SetSourceAction => ({
   },
 })
 
-export const setNamespace = (namespace: Namespace): SetNamespaceAction => ({
-  type: ActionTypes.SetNamespace,
-  payload: {
-    namespace,
-  },
-})
+export const executeHistogramQuery = () => async (
+  dispatch,
+  getState: GetState
+) => {
+  const state = getState()
+  const queryConfig = getDeep<QueryConfig | null>(
+    state,
+    'logs.histogramQueryConfig',
+    null
+  )
+  const timeRange = getDeep<TimeRange | null>(state, 'logs.timeRange', null)
+  const namespace = getDeep<Namespace | null>(
+    state,
+    'logs.currentNamespace',
+    null
+  )
+  const proxyLink = getDeep<string | null>(
+    state,
+    'logs.currentSource.links.proxy',
+    null
+  )
+
+  if (queryConfig && timeRange && namespace && proxyLink) {
+    const query = buildQuery(timeRange, queryConfig)
+    const response = await executeQueryAsync(proxyLink, namespace, query)
+
+    dispatch({
```

Comment:
could you factor this out into an action and dispatch via setHistogramData action creator directly?